deploy-review/solutions/microservice-session/src/test/java/com/redhat/training/conference/session/SessionResourceTest.java

```diff
@@ -0,0 +1,57 @@
+package com.redhat.training.conference.session;
+
+import org.junit.jupiter.api.*;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.junit.jupiter.api.Test;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+
+import javax.inject.Inject;
+import com.redhat.training.conference.speaker.SpeakerService;
+
+@QuarkusTest
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class SessionResourceTest {
+
+    @RestClient
+    @Inject
+    @InjectMock
+    SpeakerService speakerService;
+
+    @Test
+    @Order(1)
+    public void checkListOfSessionsIsEmpty() {
+        given()
+        .when()
+            .get("/sessions")
+        .then()
+            .statusCode(200)
+            .body("$.size()", is(0));
+    }
+
+    @Test
+    @Order(2)
+    public void testCreateSession () {
+        given()
+            .contentType("application/json")
+            .and()
+            .body(sessionWithSpeakerId(12))
+        .when()
+            .post("/sessions")
+        .then()
+            .statusCode(201)
```

Comment:
What about validating the header you are declaring in the openapi of the endpoint?

Reply:
I am no more using the headers as there are lot of changes that need to be done. So no more validation is required.